pmlc/dialect/pxa/tests/stencil.mlir

```diff
@@ -47,7 +47,37 @@ module {
   // CHECK: affine.parallel
   // CHECK-NOT: is_gemm
 
-// CHECK-LABEL: @gemm_operation_rewrite_i32
+  // CHECK-LABEL: @no_gem_no_stride_one_1
+  func @no_gem_no_stride_one_1(%arg0: memref<100x100xf32>, %arg1: memref<100x100xf32>, %arg2: memref<100x100xf32>) {
+    affine.parallel (%i, %j, %k) = (0, 0, 0) to (100, 100, 100) {
+      %0 = affine.load %arg1[%k, %i] : memref<100x100xf32>
+      %1 = affine.load %arg0[%k, %j] : memref<100x100xf32>
+      %3 = addf %0, %1 :f32
+      %2 = mulf %0, %3 : f32
+      pxa.reduce add %2, %arg2[%i, %j] : memref<100x100xf32>
+      "affine.terminator"() : () -> ()
+    }
+    return
+  }
+  // CHECK: affine.parallel
+  // CHECK-NOT: is_gemm
+
+  // CHECK-LABEL: @no_gem_no_stride_one_2
+  func @no_gem_no_stride_one_2(%arg0: memref<100x100xf32>, %arg1: memref<100x100xf32>, %arg2: memref<100x100xf32>) {
+    affine.parallel (%i, %j, %k) = (0, 0, 0) to (100, 100, 100) {
+      %0 = affine.load %arg1[%k, %i] : memref<100x100xf32>
+      %1 = affine.load %arg0[%k, 2*%j] : memref<100x100xf32>
+      %3 = addf %0, %1 :f32
+      %2 = mulf %0, %3 : f32
+      pxa.reduce add %2, %arg2[%i, %j] : memref<100x100xf32>
+      "affine.terminator"() : () -> ()
+    }
+    return
+  }
+  // CHECK: affine.parallel
+  // CHECK-NOT: is_gemm
```

Comment:
Can we use some other approach rather than adding an attribute? I don't think this is a good long-term approach. Since this is only for testing, you could use a remark or just actually lower this to a `xsmm.gemm` op.

Reply:
We probably will end up there. Once the XSMM is integrated. This is code that is needed to do that. I expect these tests to change.

Reply:
SGTM